framework/src/mfem/problem/MFEMProblem.C

```diff
@@ -321,14 +327,17 @@ MFEMProblem::addFunction(const std::string & type,
   // are only of space or only of time.
   if (std::find(SCALAR_FUNCS.begin(), SCALAR_FUNCS.end(), type) != SCALAR_FUNCS.end())
   {
-    _scalar_functions[name] = makeScalarCoefficient<mfem::FunctionCoefficient>(
+    getCoefficients().declareScalar<mfem::FunctionCoefficient>(
+        name,
+
         [&func](const mfem::Vector & p, double t) -> mfem::real_t
         { return func.value(t, pointFromMFEMVector(p)); });
```

Comment:
This answers my question about whether MFEM coefficients are on-the-fly/functors. They clearly are. Very cool